src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2640,7 +2640,9 @@ private void findSocketUsingThreading(LinkedList<Inet6Address> inetAddrs,
             int portNumber,
             int timeoutInMilliSeconds) throws IOException, InterruptedException {
         assert timeoutInMilliSeconds != 0 : "The timeout cannot be zero";
-        assert inetAddrs.isEmpty() == false : "Number of inetAddresses should not be zero in this function";
+        
```

Comment:
I don't see here any value addition due to using boolean empty. 

As per my understanding SonarQube flags this because if assertion is **not enabled** then whole condition will be omit. 
Now question : If assertions are not enabled is it ok to skip this line from execution? If yes then our previous implementation was right.

Think of imaginary scenario. -> we are removing objects from list. 

**Wrong Assert**
```
assert myList.remove(myObj);
```

**Right Assert**
```
boolean objRemoved = myList.remove(myObj);
assert objRemoved;
```

In first case `myList.remove(myObj)` will be omitted if assertions are not enabled.